lib/fetch/util.js

```diff
@@ -739,18 +739,14 @@ const esIteratorPrototype = Object.getPrototypeOf(Object.getPrototypeOf([][Symbo
 
 /**
  * @see https://webidl.spec.whatwg.org/#dfn-iterator-prototype-object
- * @param {() => unknown} iterator
  * @param {string} name name of the instance
- * @param {'key'|'value'|'key+value'} kind
+ * @param {symbol} kInternalIterator
  * @param {string | number} [keyIndex]
  * @param {string | number} [valueIndex]
  */
-function makeIterator (iterator, name, kind, keyIndex = 0, valueIndex = 1) {
-  const object = {
-    index: 0,
-    kind,
-    target: iterator
-  }
+function createIterator (name, kInternalIterator, keyIndex = 0, valueIndex = 1) {
+  const kInternalObject = Symbol('internal Object')
```

Comment:
Is there a reason why you need a new Symbol for every new created iterator?

Reply:
If they were the same, the `FormData` Iterator would be able to call the `Headers` Iterator.
```js
Object.getPrototypeOf(new FormData().keys()).next.call(new Headers([["a", "b"]]).keys()); 
```

Reply:
is there a perf gain when avoiding the symbol generation by creating a createFormDataIterator and a createheadersIterator?

Reply:
No, this symbol is not a performance bottleneck because it is only called when the interface is created, and therefore does not improve it.